cortex-mixin/recording_rules.libsonnet

```
@@ -69,12 +69,21 @@ local utils = import 'mixin-utils/utils.libsonnet';
         rules: [
           {
             // Convenience rule to get the number of replicas for both a deployment and a statefulset.
+            // Multi-zone deployments are grouped together removing the "zone-X" suffix.
             record: 'cluster_namespace_deployment:actual_replicas:count',
             expr: |||
-              sum by (cluster, namespace, deployment) (kube_deployment_spec_replicas)
-                or
               sum by (cluster, namespace, deployment) (
-                label_replace(kube_statefulset_replicas, "deployment", "$1", "statefulset", "(.*)")
+                label_replace(
+                  kube_deployment_spec_replicas,
+                  "deployment", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"
```

Why is the first question mark necessary? Shouldn't it be:

```suggestion
                  "deployment", "$1", "deployment", "(.*)(?:-zone-[a-z])?"
```

---

The first question mark is to make it non-greedy. Since the `(?:-zone-[a-z])?` is optional (ending `?`), if the first `(.*)` is greedy then it always match everything and never removes the zone. Adding `(.*?)` we make the first `.*` non greedy.

---

TIL! Can you add a comment to this effect please?

---

Sure, done.